test/surrogate.test.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const test = require('tap').test
+const validator = require('is-my-json-valid')
+const build = require('..')
+
+test(`render a string with surrogate pairs as JSON:test 1`, (t) => {
+  t.plan(2)
+
+  const schema = {
+    title: 'surrogate',
+    type: 'string'
+  }
+
+  const validate = validator(schema)
+  const stringify = build(schema)
+  const output = stringify('𝌆')
+
+  t.equal(output, '"𝌆"')
+  t.ok(validate(JSON.parse(output)), 'valid schema')
+})
+
+test(`render a string with surrogate pairs as JSON: test 2`, (t) => {
+  t.plan(2)
+
+  const schema = {
+    title: 'long',
+    type: 'string'
+  }
+
+  const validate = validator(schema)
+  const stringify = build(schema)
+  const output = stringify('\uD834\uDF06')
+
+  t.equal(output, '"𝌆"')
+  t.ok(validate(JSON.parse(output)), 'valid schema')
+})
+
+test(`render a string with Unpaired surrogate code as JSON`, (t) => {
+  t.plan(1)
+
+  const schema = {
+    title: 'surrogate',
+    type: 'string'
+  }
+
+  const validate = validator(schema)
+  const stringify = build(schema)
+  const output = stringify('\uDF06\uD834')
+  // t.equal(output, '"\\udf06\\ud834"')
```

Comment:
See above. This doesn’t match native `JSON.stringify` output. You can use [V8 v7.2 / Chrome 72](https://v8.dev/blog/v8-release-72#well-formed-json.stringify) to compare your implementation.

Reply:
![image](https://user-images.githubusercontent.com/7828671/54869689-e3f0ac00-4db9-11e9-9b19-d7e91006fa35.png)
This is snapshot from chrome devtools.
My chrome is up to date
![image](https://user-images.githubusercontent.com/7828671/54869702-34680980-4dba-11e9-8207-9a875d585210.png)

Reply:
Oh, I see, I saw the two surrogates in the line I commented on and assumed it was a surrogate pair, while in fact they’re two lone surrogates. My bad!